src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -171,12 +183,42 @@ public User getUserObject() {
         return this.user;
     }
 
+    @Inject(method = "handleClientSettings", at = @At("RETURN"))
```

Comment:
Should be at HEAD because the event should fire _before_ the player object gets updated